src/main/java/io/quarkus/search/app/quarkusio/QuarkusIO.java

```diff
@@ -81,107 +84,243 @@ public static Path yamlQuarkiverseMetadataPath(String version) {
+
+        validateRepositories(mainRepository, localizedSites, failureCollector);
+    }
+
+    private static void validateRepositories(
+            GitCloneDirectory mainRepository,
+            Map<Language, GitCloneDirectory> localizedSites,
+            FailureCollector failureCollector) {
+        ObjectId latestMainRev = mainRepository.currentSourcesLatestHash();
+        for (Map.Entry<Language, GitCloneDirectory> localized : localizedSites.entrySet()) {
+            localized.getValue().currentUpstreamSubmoduleSourcesHash().ifPresentOrElse(commitHash -> {
+                if (latestMainRev.equals(commitHash)) {
+                    // means that the localized repo points to the latest main one and we can ignore further checks
+                    return;
+                }
+
+                // Otherwise let's see when the localized site was last synced,
+                // by looking at when that latest commit was committed:
+                Instant lastSync = GitUtils.committedInstant(mainRepository.git(), commitHash);
+                // Sync happens each week, so we can have
+                //  1. a commit to main repo on Monday
+                //  2. no further commits during the week
+                //  3. sync on the weekend (Sunday?)
+                //  4. commits to main repo
+                //  at this point the difference between latest commit in localized site and current date can be
+                //  more than 7 days; so we give it 2 weeks period before we start considering that there's a sync problem:
+                Duration duration = Duration.between(lastSync, Instant.now());
+                if (duration.compareTo(Duration.ofDays(14)) > 0) {
+                    failureCollector.warning(
+                            FailureCollector.Stage.TRANSLATION,
+                            "Localized repository '" + localized.getKey() + "' is out of sync for " + duration);
+                }
+            }, () -> {
+                Log.debug(
+                        "Repository for a localized site '" + localized.getKey()
+                                + "' does not have an \"upstream\" directory. Site will be considered up-to-date. ");
+            });
+        }
     }
 
     @Override
     public void close() throws Exception {
         try (var closer = new Closer<Exception>()) {
             closer.push(CloseableDirectory::close, prefetchedGuides);
-            closer.push(GitCloneDirectory::close, mainRepository);
-            closer.pushAll(GitCloneDirectory::close, localizedSites.values());
+            closer.pushAll(GitCloneDirectory::close, allSites.values());
         }
     }
 
     public Stream<Guide> guides() throws IOException {
-        return Stream.concat(versionedGuides(), legacyGuides());
+        return Stream.concat(
+                Stream.concat(versionedGuides(), legacyGuides()),
+                Stream.concat(quarkiverseGuides(), legacyQuarkiverseGuides()));
     }
```

Comment:
I'm seeing 7032 indexed documents using your branch vs. 6348 using the main branch.

I suspect some Quarkiverse guides are now getting duplicated... Did you take into account the fact that some `quarkus.yaml` files (the legacy ones, I think? not sure) contain reference to Quarkiverse guides, which should be handled as such? That was the whole point of having a `createGuide()` method that would delegate to either `createCoreGuide` or `createQuarkiverseGuide` depending on the origin.

I'm not sure this will work... This file may contain quarkiverse entries, no?

Reply:
yeah ... I haven't tracked down where it comes from yet ... and yes.... the idea here was to make one pass over legacy guides and skip the Quarkiverse guides, and then do them in the second pass and have a single Quarkiverse doc (so from the idea point that should've worked but impl part needs more attention I guess 😃)